src/interfaces/common.ts

```diff
@@ -79,29 +87,30 @@ interface ILogger {
  * */
 interface IAppDelegate {
     moddleOptions;
-    getServicesProvider(IExecution): any;       // to respond to all named services
+    /**
+     * Get the service task handlers, default to `this`, so you can add handlers on this class directly.
+     */
+    getServicesProvider(execution: IExecution): IServiceProvider | Promise<IServiceProvider>;
     sendEmail(to, msg, body);
-    executionStarted(execution);
+    executionStarted(execution: IExecution);
     startUp(options); // start of server
-    messageThrown(signalId, data, messageMatchingKey: any, item: IItem);
-    signalThrown(signalId, data, messageMatchingKey: any, item: IItem);
+    messageThrown(signalId: string, data, messageMatchingKey: any, item: IItem);
+    signalThrown(signalId: string, data, messageMatchingKey: any, item: IItem);
     /**
      * 
      * is called when an event throws a message that can not be answered by another process
      * 
      * @param messageId
      * @param data
      */
-    issueMessage(messageId, data);
-    issueSignal(messageId, data);
+    issueMessage(messageId: string, data);
+    issueSignal(messageId: string, data);
     /**
      * is called only if the serviceTask has no implementation; otherwise the specified implementation will be called.
      * 
      * @param item
      */
-    serviceCalled(serviceName,data,item: IItem);
-
-
+    serviceCalled(input: Record<string, unknown>, execution: IExecution, item: IItem): unknown;
```

Comment:
2. And you can get name from `(item.node as ServiceTask).serviceName`, from the `item: IItem`.